File: ModuleDaily/src/main/java/com/qihuan/dailymodule/contract/DailyContract.java
package com.qihuan.dailymodule.contract;

import com.qihuan.commonmodule.base.BasePresenter;
import com.qihuan.commonmodule.base.BaseView;
import com.qihuan.dailymodule.model.bean.DailyItemBean;
import com.qihuan.dailymodule.model.bean.TopStoryBean;

import java.util.List;

public interface DailyContract {
    interface View extends BaseView {
        void latestDaily(List<TopStoryBean> topList);

        void beforeDaily(boolean isRefresh, List<DailyItemBean> dailyList);

        void onRefreshEnd(boolean success);

        void onLoadMoreEnd(boolean success);
    }

    interface Presenter extends BasePresenter<View> {
        void getLatestDaily();

        void getBeforeDaily();
    }
}

File: ModuleDaily/src/main/java/com/qihuan/dailymodule/contract/DailyContract.kt
package com.qihuan.dailymodule.contract

import com.qihuan.commonmodule.base.BasePresenter
import com.qihuan.commonmodule.base.BaseView
import com.qihuan.dailymodule.model.bean.DailyItemBean
import com.qihuan.dailymodule.model.bean.TopStoryBean

interface DailyContract {
    interface View : BaseView {
        fun latestDaily(topList: List<TopStoryBean>)

        fun beforeDaily(isRefresh: Boolean, dailyList: List<DailyItemBean>)

        fun onRefreshEnd(success: Boolean)

        fun onLoadMoreEnd(success: Boolean)
    }

    interface Presenter : BasePresenter<View> {
        fun getLatestDaily()

        fun getBeforeDaily()
    }
}
